src/native/libs/System.Security.Cryptography.Native/pal_evp_pkey.c

```diff
@@ -109,6 +109,37 @@ int32_t CryptoNative_EvpPKeyType(EVP_PKEY* key)
     }
 }
 
+int32_t CryptoNative_EvpPKeyFamily(const EVP_PKEY* key)
+{
+    int32_t base_id = EVP_PKEY_get_base_id(key);
+    switch (base_id)
+    {
+        case EVP_PKEY_RSA_PSS:
+        case EVP_PKEY_RSA:
+            return PalPKeyFamilyId_RSA;
+        case EVP_PKEY_EC:
+            return PalPKeyFamilyId_ECC;
+        case EVP_PKEY_DSA:
+            return PalPKeyFamilyId_DSA;
+        default:
+            break;
+    }
+
+#ifdef NEED_OPENSSL_3_0
+    if (API_EXISTS(EVP_PKEY_is_a))
+    {
+        ERR_clear_error();
+
+        if (EVP_PKEY_is_a(key, "ML-KEM-512") || EVP_PKEY_is_a(key, "ML-KEM-768") || EVP_PKEY_is_a(key, "ML-KEM-1024"))
```

Comment:
This is going to look awfully icky when SLH-DSA gets here and adds its 9000 type tests.

I don't have a better idea offhand, so it's just a thing to keep in mind.

Reply:
@bartonjs 

If `EVP_PKEY_get_base_id` does something sane for SLH-DSA / ML-DSA then we can avoid the is_a. It might be better for signature algorithms since that is all established things in OpenSSL.

So, we shouldn’t jump to doing what ML-KEM is doing without understanding what `EVP_PKEY_get_base_id` does for the other algorithms, first.